package-dev/Plugins/macOS/SentryNativeBridge.m

```diff
@@ -0,0 +1,246 @@
+#import <Foundation/Foundation.h>
+#include <dlfcn.h>
+
+static int loadStatus = -1; // unitialized
+
+static Class SentrySDK;
+static Class SentryScope;
+static Class SentryBreadcrumb;
+static Class SentryUser;
+
+#define LOAD_CLASS_OR_BREAK(name)                                                                  \
+    name = (__bridge Class)dlsym(dylib, "OBJC_CLASS_$_" #name);                                    \
+    if (!name) {                                                                                   \
+        NSLog(@"Sentry (native bridge): Couldn't load %@ class from the dynamic library", name);   \
+        break;                                                                                     \
+    }
+
+// Returns (bool): 0 on failure, 1 on success
+// WARNING: you may only call other Sentry* functions AFTER calling this AND only if it returned "1"
+int SentryNativeBridgeLoadLibrary()
+{
+    if (loadStatus == -1) {
+        loadStatus = 0; // init to "error"
+        do {
+            void *dylib = dlopen("@executable_path/../PlugIns/Sentry.dylib", RTLD_LAZY);
+            if (!dylib) {
+                NSLog(@"Sentry (native bridge): Couldn't load Sentry.dylib - dlopen() failed");
+                break;
+            }
+
+            LOAD_CLASS_OR_BREAK(SentrySDK)
+            LOAD_CLASS_OR_BREAK(SentryScope)
+            LOAD_CLASS_OR_BREAK(SentryBreadcrumb)
+            LOAD_CLASS_OR_BREAK(SentryUser)
+
+            // everything above passed - mark as successfully loaded
+            loadStatus = 1;
+        } while (false);
+    }
+    return loadStatus;
+}
+
+const void *SentryNativeBridgeOptionsNew()
+{
+    return CFBridgingRetain([[NSMutableDictionary alloc] init]);
```

Comment:
There's no ARC available? I haven't seen `CFBridgingRetain` used with ARC so wondering why we need this here

Reply:
CFBridgingRetain is used specifically with ARC - I had to change that because of some compilation issues. I'm glad if someone with good objective-c understanding has a look but to my limited knowledge, this should be OK - we want to take ownership of the pointer here, keeping the value in memory, later in `SentryNativeBridgeStartWithOptions` it's transferred back and left to ARC to free as needed.